diffkemp/llvm_ir/build_llvm.py

```diff
@@ -124,6 +131,61 @@ def _get_kernel_tar_from_upstream(self):
 
         return tarname
 
+    def _get_kernel_tar_from_centos(self):
+        """
+        Download sources of the required kernel from the CentOS Git.
+        First the corresponding Git repository is downloaded and the URLs of the
+        necessary files are derived from it. Then it downloads them and extracts
+        the kernel source.
+        :returns Name of the tar file containing the sources.
+        """
+        centos_git_url = "https://git.centos.org/git/rpms/kernel.git"
+        self.configfile = "kernel-{}-x86_64.config".format(
+            self.kernel_version.split("-")[0])
+
+        tmpdir = mkdtemp()
+        cwd = os.getcwd()
+        os.chdir(tmpdir)
+
+        # Clone the Git repository and checkout the appropriate commit.
+        check_call(["git", "clone", "-q", centos_git_url])
+        os.chdir("kernel")
+        check_call(["git", "checkout", "-q",
+            "tags/imports/c7/kernel-{}.el7".format(self.kernel_version)])
+
+        # Get the hashes of the required files and the config from the metadata
+        # file
+        filelist = []
+        with open(".kernel.metadata", 'r') as metadata:
+            for line in metadata:
+                filelist.append(line.rstrip().split(' '))
+
+        shutil.copy("SOURCES/" + self.configfile, self.kernel_base_path)
+        os.chdir(self.kernel_base_path)
+        shutil.rmtree(tmpdir)
+
+        # Download the actual sources
+        for entry in filelist:
+            url = "https://git.centos.org/sources/kernel/c7/{}".format(entry[0])
+            filename = entry[1].split("/")[1]
+
+            urlretrieve(url, os.path.join(self.kernel_base_path, filename),
+                    show_progress)
```

Comment:
Wouldn't it be better to clone the centos-common-git repo (https://git.centos.org/git/centos-git-common.git) and run the get_sources.sh script from there? It does the same as you do and if they change the way the files are downloaded, it will keep working.